### PR TITLE
Test API expansion

### PR DESCRIPTION
**Pull request by fokinv**
Moved the test function into a new file, made a macro for setting the attributes.

---

- [x] `./mach build -d` does not report any errors
- [x] `./mach test-tidy` does not report any errors
- [ ] These changes fix #__ (github issue number if applicable).

- [ ] There are tests for these changes OR
- [ ] These changes do not require tests because _____

### CODE REVIEW

File: components/net/bluetooth_test.rs [hunk 1]
```diff
@@ -0,0 +1,387 @@
+/* This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at http://mozilla.org/MPL/2.0/. */
+
+use bluetooth_thread::BluetoothManager;
+use device::bluetooth::BluetoothDevice;
+use device::bluetooth::BluetoothGATTCharacteristic;
+use device::bluetooth::BluetoothGATTDescriptor;
+use device::bluetooth::BluetoothGATTService;
```
Comment (zakorgy) [inline components/net/bluetooth_test.rs]: These uses could be written in two or maybe one line(s).

File: components/net/bluetooth_test.rs [hunk 2]
```diff
@@ -0,0 +1,387 @@
...
+use std::string::String;
+
+const ADAPTER_ERROR: &'static str = "No adapter found";
+
+const WRONG_DATA_SET_ERROR: &'static str = "Wrong data set name was provided";
+
+const FAILED_SET_ERROR: &'static str = "Failed to set an attribute for testing";
+
+macro_rules! set_attribute_or_return_error(
+    ($function:expr, $sender:expr) => (
```
Comment (zakorgy) [inline components/net/bluetooth_test.rs:19]: Can we omit the `$sender:expr`?

File: components/net/bluetooth_test.rs [hunk 3]
```diff
@@ -0,0 +1,387 @@
...
+        match $function {
+            Ok(_) => (),
+            Err(_) => return drop($sender.send(Err(BluetoothError::Type(FAILED_SET_ERROR.to_string())))),
```
Comment (zakorgy) [inline components/net/bluetooth_test.rs:22]: And write here just `sender` instead of `$sender`.

File: components/net/bluetooth_test.rs (hunk 2 above)
Comment (fokinv) [reply components/net/bluetooth_test.rs:19]: No can do, Sir!

File: components/net/bluetooth_test.rs (hunk 3 above)
Comment (fokinv) [reply components/net/bluetooth_test.rs:22]: I said, No can do, Sir!

**Pull request merged**